File: backend/app/models/meeting.py
"""Meeting model"""
from sqlalchemy import Column, String, DateTime, ForeignKey, Enum, Boolean, Text
from sqlalchemy.sql import func
from app.core.database import Base
import enum


class MeetingStatusEnum(str, enum.Enum):
    """Meeting status"""
    DRAFT = "draft"
    LIVE = "live"
    COMPLETED = "completed"
    ARCHIVED = "archived"


class Meeting(Base):
    """Meeting model"""
    __tablename__ = "meetings"

    id = Column(String(36), primary_key=True, index=True)
    organization_id = Column(String(36), ForeignKey("organizations.id"), nullable=False, index=True)
    created_by = Column(String(36), ForeignKey("users.id"), nullable=False)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    status = Column(Enum(MeetingStatusEnum), default=MeetingStatusEnum.DRAFT, nullable=False, index=True)
    started_at = Column(DateTime(timezone=True), nullable=True)
    ended_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    # No special table args required currently
    __table_args__ = ()

File: frontend/src/store/meeting.ts
import { create } from "zustand";

export interface Meeting {
  id: string;
  organization_id: string;
  title: string;
  description?: string;
  status: "draft" | "live" | "completed" | "archived";
  created_at: string;
  updated_at: string;
}

export interface TranscriptChunk {
  id: string;
  speaker?: string;
  text: string;
  timestamp?: number;
  sequence_number: number;
}

interface MeetingStore {
  currentMeeting: Meeting | null;
  transcriptChunks: TranscriptChunk[];
  setCurrentMeeting: (meeting: Meeting) => void;
  addTranscriptChunk: (chunk: TranscriptChunk) => void;
  clearMeeting: () => void;
}

export const useMeetingStore = create<MeetingStore>((set) => ({
  currentMeeting: null,
  transcriptChunks: [],
  setCurrentMeeting: (meeting) => set({ currentMeeting: meeting }),
  addTranscriptChunk: (chunk) =>
    set((state) => ({
      transcriptChunks: [...state.transcriptChunks, chunk],
    })),
  clearMeeting: () => set({ currentMeeting: null, transcriptChunks: [] }),
}));
